src/main/java/org/graylog/plugins/usagestatistics/dto/Os.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (C) 2015 Graylog, Inc. (hello@graylog.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.graylog.plugins.usagestatistics.dto;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+
+@JsonAutoDetect
+@AutoValue
+public abstract class Os {
+    public static Os create(String name,
+                            String version,
+                            String arch,
+                            String platformName) {
+        return new AutoValue_Os(name, version, arch, platformName);
```

Comment:
If the code is being run on a yet unknown OS, the structure will be populated with the contents of the os.name, os.version and os.arch system properties.